File: Semestr 6/JavaProjekt/Projekt/src/main/java/com/example/projekt/models/Printer.java
package com.example.projekt.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Printer {
    @Id
    @GeneratedValue
    private int id;

    private String model;
    private String manufacturer;
    private String serialNumber;
    private String invoiceDescription;
    private String description;
    private String contractNumber;
    private String ip;
    private String department;

    private boolean test;

}

File: Semestr 6/JavaProjekt/Frotnend/src/app/Models/Printer.ts
import { Department } from "./Department";


export interface Printer {
    id: number;
    model: string;
    manufacturer: string;
    description: string | null;
    invoiceDescription: string | null;
    contractNumber: string | null;
    iP: string | null;
    test: boolean;
    department: Department;
    serialNumber: string;
}